src/coap_subscribe.c

```diff
@@ -340,6 +351,8 @@ coap_op_observe_read(FILE *fp, coap_subscription_t **observe_key,
     return 1;
   }
 fail:
+  /* CID 1566474 */
+  coap_delete_binary(scratch);
   return 0;
```

Comment:
For safety, I suggest `*raw_packet = NULL` as well.

Reply:
Corrected